burn-core/src/nn/sin_embedding.rs

```diff
@@ -0,0 +1,198 @@
+use alloc::vec::Vec;
+
+use crate as burn;
+use crate::config::Config;
+use crate::module::Module;
+use crate::tensor::backend::Backend;
+use crate::tensor::Tensor;
+use burn_tensor::Data;
+
+use libm::{cosf, expf, logf, sinf};
+
+/// Configuration to create an [SinEmbedding](SinEmbedding) layer.
+#[derive(Config)]
+pub struct SinEmbeddingConfig {
+    /// The number of embedding vectors.
+    n_embedding: usize,
+
+    /// The size of each vector.
+    d_model: usize,
+
+    /// Max time scale to use.
+    #[config(default = "10_000")]
+    max_timescale: usize,
+}
+
+/// Lookup table to store sinusoidal vectors for positional embedding.
+#[derive(Module, Debug)]
+pub struct SinEmbedding<B: Backend> {
+    sinusoids: Tensor<B, 3>,
+}
+
+impl SinEmbeddingConfig {
+    /// Initialize a new [SinEmbedding](SinEmbedding) module.
+    pub fn init<B: Backend>(&self) -> SinEmbedding<B> {
+        let sinusoids = generate_sinusoids::<B>(self.n_embedding, self.d_model, self.max_timescale)
+            .unsqueeze::<3>();
+
+        SinEmbedding { sinusoids }
+    }
+}
+
+impl<B: Backend> SinEmbedding<B> {
+    /// Applies the forward pass on the input tensor by adding the sinusoids to the input.
+    ///
+    /// # Shapes
+    ///
+    /// - input: [batch_size, seq_length, d_model]
+    /// - output: [batch_size, seq_length, d_model]
+    pub fn forward(&self, input: Tensor<B, 3>) -> Tensor<B, 3> {
+        input.add(self.sinusoids.clone())
```

Comment:
I think a slice is normally needed here. See https://pytorch.org/tutorials/beginner/transformer_tutorial.html for reference.

Reply:
I often see dropout as well